wholegames-intro.Rmd

```diff
@@ -0,0 +1,146 @@
+#  (PART) Whole Game(s) {-}
+
+# Introduction
+
+Similar to the [Whole Game chapter in the R packages book by Hadley Wickham and Jenny Bryan](https://r-pkgs.org/whole-game.html), we shall go through how to add HTTP tests to a minimal package.
+However, we will do it _three_ times to present alternative approaches: with [vcr](#vcr), [httptest](#httptest), [webfakes](#webfakes).
+After that exercise, we shall compare approaches: we will compare both packages that involve mocking i.e. [vcr vs. httptest](#mocking-pkgs-comparison); and all three HTTP packages in a [last chapter](#pkgs-comparison).
+The next section will then present single topics such as "how to deal with authentication" in further details.
+
+## Our example package
+
+Our minimal package, [`exemplighratia`](https://github.com/maelle/exemplighratia), accesses the GitHub status API and one endpoint of GitHub V3 REST API.
+It is named after the Latin phrase _exempli gratia_ that means "for instance", with an H for GH.
+If you really need to interact with GitHub V3 API, we recommend the [gh package](https://gh.r-lib.org/).
+We also recommend looking at the source of the gh package, and at the docs of GitHub V3 API, in particular about [authentication](https://developer.github.com/v3/#authentication).
+
+GitHub V3 API works without authentication too, but at a lower rate.
+For the sake of having an example of a package _requiring_ authentication we shall assume the API is not usable without authentication.
+Authentication is the setting of a token in a HTTP header, so no OAuth dance or so. 
+
+GitHub Status API, on the contrary, does not necessitate authentication at all.
+
+So we have two functions, one that works without authentication, one that works with authentication.
```

Comment:
instead of "we have", maybe "we will make"? "we have" suggests you've already shown the two functions